techtv2ovs/utils.py

```diff
@@ -0,0 +1,302 @@
+        s3client = boto3.client('s3')
+        s3list = s3client.list_objects_v2(Bucket=TTV_THUMB_BUCKET, Prefix=ttv_video.external_id)
+        if 'Contents' not in s3list or not s3list['Contents']:
+            ttv_video.thumbnail_status = ImportStatus.MISSING
+            ttv_video.save()
+            return
+        user_list = [thumb for thumb in s3list['Contents'] if '{}/jumbo'.format(ttv_video.ttv_id) in thumb['Key']]
+        default_list = [thumb for thumb in s3list['Contents'] if 'thumbnails/jumbo' in thumb['Key']]
+        idx = 0
+        for thumb in user_list + default_list:
+            new_key = thumb['Key'].split('/')[-1].replace('jumbo', str(idx))
+            try:
+                src = {'Bucket': TTV_THUMB_BUCKET, 'Key': thumb['Key']}
+                dst = {
+                    'Bucket': settings.VIDEO_S3_THUMBNAIL_BUCKET,
+                    'Key': 'thumbnails/techtv/{}/{}'.format(ttv_video.video.hexkey, new_key)
+                }
+
+                if self.aws:
+                    s3client.copy(src, dst['Bucket'], dst['Key'])
+
+                VideoThumbnail.objects.get_or_create(s3_object_key=dst['Key'], defaults={
+                    'bucket_name': dst['Bucket'],
+                    'video': ttv_video.video
+                })
+                idx += 1
+            except Exception as exc:
+                self.output.write("Error: thumbnail {} for ttv video {}".format(src['Key'], ttv_video.ttv_id))
+                ttv_video.status = ImportStatus.ERROR
+                ttv_video.thumbnail_status = ImportStatus.ERROR
+                ttv_video.errors += '{}\n\n'.format(str(exc))
+
+        if ttv_video.thumbnail_status != ImportStatus.ERROR:
+            ttv_video.thumbnail_status = ImportStatus.COMPLETE
+        ttv_video.save()
+
+    def process_files(self, ttv_video, s3videos):
+        """
+        Import videofiles, thumbnails, and captions for a TechTVVideo.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo object to process
+            s3videos (list): List of video file S3 objects
+        """
+        self.process_thumbs(ttv_video)
+        process_videofiles.delay(ttv_video.id, s3videos, self.aws)
+
+    def process_videos(self, ttvcollection):
+        """
+        Process all the videos for a TechTV Collection
+
+        Args:
+            ttvcollection (TechTVCollection): a TechTVCollection object
+        """
+        video_query = (
+            'SELECT v.id, v.title, v.description, v.external_id, v.private, v.private_token '
+            'from videos v inner join collection_videos cv on cv.video_id = v.id where cv.collection_id = {} '
+            'and v.status = \'approved\' AND v.file_restored = 1 AND v.locked=0;'.format(ttvcollection.id)
+        )
+        videos = mysql_query(video_query, self.connection)
+        for video in videos:
+            (v_id, v_title, v_description, external_id, private, token) = video
```

Comment:
You can do this unpacking in the for loop directly:

    for v_id, v_title, v_description, external_id, private, token in videos:

Reply:
✅